src/solutions/ode_solutions.jl

```diff
@@ -15,7 +15,40 @@ struct ODESolution{T,N,uType,uType2,DType,tType,rateType,P,A,IType,DE} <: Abstra
   destats::DE
   retcode::Symbol
 end
-(sol::ODESolution)(t,deriv::Type=Val{0};idxs=nothing,continuity=:left) = sol.interp(t,idxs,deriv,sol.prob.p,continuity)
+
+function (sol::ODESolution)(t,deriv::Type=Val{0};idxs=nothing,continuity=:left)
+  if deriv != Val{0}
+    error("Not Implemented")
+  end
+
+  if idxs === nothing
+    if t isa Real
+      sol.interp(t,idxs,deriv,sol.prob.p,continuity)
+    else
+      augment(sol.interp(t,idxs,deriv,sol.prob.p,continuity), sol)
+    end
+  elseif idxs isa Vector{<:Integer} || idxs isa Integer
+      sol.interp(t,idxs,deriv,sol.prob.p,continuity)
+  elseif issymbollike(idxs)
+    if t isa Real
+      interp_sol = augment(sol.interp([t],nothing,deriv,sol.prob.p,continuity), sol)
```

Comment:
Can't we use

```suggestion
      interp_sol = augment(sol.interp(t,nothing,deriv,sol.prob.p,continuity), sol)
```

Reply:
We will have to add an extra constructor to `DiffEqArray` in RecursiveArrayTools.jl to support this. https://github.com/SciML/RecursiveArrayTools.jl/blob/master/src/vector_of_array.jl#L42-L46 doesn't cover the case where `A.u` is Vector{Real}.

Reply:
Will be added in https://github.com/SciML/RecursiveArrayTools.jl/pull/148

Reply:
The current API of interpolation outputs `Vector{<:Real}` when the input `t isa Real`. DiffEqArray expects a vector of times `t` (`Vector{<:Real}`)  and vector of corresponding states `u` (`Vector{Vector{<:Real}}`). 

These two factors together make it difficult to accommodate what you are suggesting without making substantial changes to interpolation and DiffEqArray indexing.